lib/avromatic/model/attributes.rb

```diff
@@ -127,24 +128,33 @@ def avro_field_class(field_type)
           when :record
             # TODO: This should add the generated model to a module.
             # A hash of generated models should be kept by name for reuse.
-            Class.new do
-              include Avromatic::Model.build(schema: field_type)
-            end
+            Avromatic::Model.model(schema: field_type)
           else
             raise "Unsupported type #{field_type}"
           end
         end
 
         def union_field_class(field_type)
-          avro_field_class(Avromatic::Model::Attributes.first_union_schema(field_type))
+          field_classes = field_type.schemas.select { |schema| schema.type_sym != :null }
+                            .map { |schema| avro_field_class(schema) }
+
+          if field_classes.size == 1
+            field_classes.first
+          else
+            Avromatic::Model::AttributeType::Union[*field_classes]
+          end
         end
 
-        def avro_field_options(field)
+        def avro_field_options(field, field_class)
           options = {}
 
-          custom_type = Avromatic.type_registry.fetch(field)
-          coercer = custom_type.deserializer
-          options[:coercer] = coercer if coercer
+          if field_class.is_a?(Class) && field_class < Avromatic::Model::AttributeType::Union
+            # TODO: pass coercer for member attributes in union
```

Comment:
Should we raise an error if there's a member attribute with a custom coercer until we add proper support for this?